backend/src/main/kotlin/org/jetbrains/kotlinconf/backend/services/NewsService.kt

```diff
@@ -2,112 +2,131 @@ package org.jetbrains.kotlinconf.backend.services
+    private suspend fun downloadNews(): List<NewsItem> {
+        val url = "https://api.github.com/repos/$repo/contents/$folder?ref=$branch"
 
         val contents: List<GitHubItem> = try {
             client.get(url) {
                 headers {
                     append("Accept", "application/vnd.github.v3+json")
                 }
             }.body()
-        } catch (e: Exception) {
-            println("Error fetching repository contents: ${e.message}")
-            emptyList()
+        } catch (cause: Throwable) {
+            log.warn("Error fetching repository contents: ${cause.message}")
+            return emptyList()
         }
 
-        // Print the files found
-        println("Files found in $org/$repo/$branch/$normalizedPath:")
-        contents.forEach {
-            println("- ${it.name} (${it.type})")
+        val urls = contents
+            .filter { it.type == "file" && it.name.endsWith(".md") }
+            .mapNotNull { it.download_url }
+
+        val result = mutableListOf<NewsItem>()
+        for (url in urls) {
+            try {
+                val content = client.get(url).body<String>()
+                val item = parseNewsItem(content)
+                result.add(item)
+            } catch (cause: Throwable) {
+                log.warn("Error downloading $url: ${cause.message}")
+            }
         }
 
-        // Download each file that passes the filter
-        var downloadCount = 0
-        contents.forEach { item ->
-            when (item.type) {
-                "dir" -> {
-                    // If it's a directory, we could recursively download, but we'll skip for now
-                    println("Skipping directory: ${item.path}")
-                }
-                "file" -> {
-                    // Check if we should download this file
-                    if (fileFilter(item)) {
-                        item.download_url?.let { url ->
-                            try {
-                                val content = client.get(url).body<String>()
-                                withContext(Dispatchers.IO) {
-                                    File("$outputDir/${item.name}").writeText(content)
-                                }
-                                println("Downloaded: ${item.path}")
-                                downloadCount++
-                            } catch (e: Exception) {
-                                println("Error downloading ${item.path}: ${e.message}")
-                            }
-                        }
-                    }
-                }
+        return result
+    }
+
+    /**
+     * Parses a Markdown file and converts it to a NewsItem object.
+     *
+     * The Markdown file should have the following format:
+     * ```
+     * id: idstring
+     * photoUrl: http://example.com/photo.jpg
+     * title: title of the page
+     * publicationDate: "2024-01-20T00:00:00"
+     * ---
```

Comment:
Frontmatter also usually have `---` at the beginning

Reply:
nice, let's have it as well